config/localTemplate.go

```diff
@@ -735,10 +737,19 @@ func (cfg Local) TxFilterCanonicalEnabled() bool {
 	return cfg.TxIncomingFilteringFlags&txFilterCanonical != 0
 }
 
-// IsGossipServer returns true if NetAddress is set and this node supposed
-// to start websocket server
+// IsGossipServer returns true if this node supposed to start websocket or p2p server
 func (cfg Local) IsGossipServer() bool {
-	return cfg.NetAddress != ""
+	return cfg.IsWsGossipServer() || cfg.IsP2PGossipServer()
+}
+
+// IsWsGossipServer returns true if a node configured to run a listening ws net
+func (cfg Local) IsWsGossipServer() bool {
+	return cfg.NetAddress != "" && !cfg.EnableP2P
```

Comment:
Should this return true if in hybrid mode? Feels like it should..(but would return false by this logic)

Reply:
It should return true in hybrid. Well, unless both EnableHybridNode AND EnableP2P are set. Right, needs correction

Reply:
[f002752](https://github.com/algorand/go-algorand/pull/6035/commits/f00275231e6ab497160f89142d8432cd06636d47)

Reply:
@algorandskiy what if hybrid mode and `enableP2P` are both enabled - wouldn't that return false here when we want true?

Reply:
discussed offline - the last comment refers to outdated code